include/pybind11/cast.h

```diff
@@ -374,11 +374,31 @@ class type_caster_generic {
     object temp;
 };
 
-/* Determine suitable casting operator */
+/** Determine suitable casting operator for pointer-or-lvalue-casting type casters.  The type caster
+ * needs to provide `operator T*()` and `operator T&()` operators.
```

Comment:
Style nitpick: could the first and second text line be aligned? I.e.:
```c++
/** 
 * Determine suitable casting operator for pointer-or-lvalue-casting type casters.  The type caster
 * needs to provide `operator T*()` and `operator T&()` operators.
 */
```
(The misalignment probably bothers me more than it should.)